modules/apps/message-boards/message-boards-test/src/testIntegration/java/com/liferay/message/boards/search/test/MBMessageSearchTest.java

```diff
@@ -75,6 +89,52 @@ public void testSearchComments() throws Exception {
 	@Override
 	@Test
 	public void testSearchCommentsByKeywords() throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				TestPropsValues.getGroupId(), TestPropsValues.getUserId());
+
+		BlogsEntry blogsEntry = BlogsTestUtil.addEntryWithWorkflow(
+			TestPropsValues.getUserId(), RandomTestUtil.randomString(), true,
+			serviceContext);
+
+		String body = RandomTestUtil.randomString();
+
+		CommentManagerUtil.addComment(
+			TestPropsValues.getUserId(), TestPropsValues.getGroupId(),
+			BlogsEntry.class.getName(), blogsEntry.getEntryId(), body,
+			serviceContext1 -> serviceContext);
+
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			TestPropsValues.getGroupId(), 0);
+
+		long messageId = CommentManagerUtil.addComment(
+			TestPropsValues.getUserId(), TestPropsValues.getGroupId(),
+			JournalArticle.class.getName(), journalArticle.getClassPK(), body,
+			serviceContext1 -> serviceContext);
+
+		SearchContext searchContext = SearchContextTestUtil.getSearchContext(
+			TestPropsValues.getUserId(), body, LocaleUtil.US);
+
+		searchContext.setAttribute(
+			Field.CLASS_NAME_ID,
+			PortalUtil.getClassNameId(JournalArticle.class));
+		searchContext.setAttribute("discussion", Boolean.TRUE);
+
+		Indexer<MBMessage> indexer = IndexerRegistryUtil.getIndexer(
+			MBMessage.class);
+
+		Hits hits = indexer.search(searchContext);
+
+		Assert.assertEquals(hits.toString(), 1, hits.getLength());
+
+		Assert.assertEquals(
+			messageId,
+			GetterUtil.getLong(
+				hits.doc(
+					0
+				).get(
+					Field.ENTRY_CLASS_PK
+				)));
```

Comment:
This kind of method chaining is not allowed. You need to introduce an intermediate variable:
```
Document document = hits.doc(0);

Asset.assertEquals(messageId, GetterUtil.getLong(document.get(...)));
```